docs/src/main/asciidoc/security-oidc-code-flow-authentication-concept.adoc

```diff
@@ -267,285 +660,167 @@ quarkus.oidc.logout.post-logout-uri-param=returnTo
+If for example your OIDC provider is Google, you will be logged out from Google and its services.
+Instead the user might just want to log out of that specific application.
+Another use case if the OIDC provider does not have a logout endpoint.
 
-One can register `@ApplicationScoped` bean which will observe important OIDC authentication events. The listener will be updated when a user has logged in for the first time or re-authenticated, as well as when the session has been refreshed. More events may be reported in the future. For example:
+You can support a local logout with the help of the <<oidc-session,OidcSession>> which only clears the local session cookie, for example:
 
-[source, java]
+[source,java]
 ----
-import jakarta.enterprise.context.ApplicationScoped;
-import jakarta.enterprise.event.Observes;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
 
-import io.quarkus.oidc.IdTokenCredential;
-import io.quarkus.oidc.SecurityEvent;
-import io.quarkus.security.identity.AuthenticationRequestContext;
-import io.vertx.ext.web.RoutingContext;
+import io.quarkus.oidc.OidcSession;
 
-@ApplicationScoped
-public class SecurityEventListener {
+@Path("/service")
+public class ServiceResource {
 
-    public void event(@Observes SecurityEvent event) {
-        String tenantId = event.getSecurityIdentity().getAttribute("tenant-id");
-        RoutingContext vertxContext = event.getSecurityIdentity().getAttribute(RoutingContext.class.getName());
-        vertxContext.put("listener-message", String.format("event:%s,tenantId:%s", event.getEventType().name(), tenantId));
+    @Inject
+    OidcSession oidcSession;
+
+    @GET
+    @Path("logout")
+    public String logout() {
+        oidcSession.logout().await().indefinitely();
+        return "You are logged out".
     }
-}
 ----
 
-=== Single-page applications
+[[oidc-session]]
+====== Using OidcSession for local logout
 
-Check if implementing SPAs the way it is suggested in the xref:security-oidc-bearer-token-authentication-concept.adoc#single-page-applications[Single-page Applications for Service Applications] section can meet your requirements.
+`io.quarkus.oidc.OidcSession` is a wrapper around the current `IdToken`. It can help to perform a <<local-logout, Local logout>>, retrieve the current session's tenant identifier and check when the session will expire. More useful methods will be added to it over time.
+[[session-management]]
 
-If you prefer to use SPA and JavaScript API such as `Fetch` or `XMLHttpRequest`(XHR) with Quarkus web applications, be aware that OpenID Connect Providers may not support CORS for Authorization endpoints where the users are authenticated after a redirect from Quarkus. This will lead to authentication failures if the Quarkus application and the OpenID Connect Provider are hosted on the different HTTP domains/ports.
+==== Expiration via Session management
```

Comment:
```suggestion
==== Expiration through session management
```

Reply:
@michelle-purcell I did not notice Emmanuel changed this section title, it should remain as `Session management`, adding `Expiration` in front introduces a new meaning

Reply:
Fixed :+1: